packages/react-shortcuts/src/ShortcutManager/ShortcutManager.tsx

```diff
@@ -61,6 +61,22 @@ export default class ShortcutManager {
     }
   };
 
+  private allModifiersAreHeld = (heldKeys: HeldKey, event: KeyboardEvent) => {
+    // eslint-disable-next-line no-warning-comments
+    // TODO: TypeScript complains that `keyGroupIsHeld(heldKeys)`
+    // could still contain `ModifierKey[]`, which is incorrect...
+    // Until this can be resolved, we need to set type to `any` instead of `ModiferKey[]`
+    function keyGroupIsHeld(keyGroup: any) {
```

Comment:
If I leave this as `ModiferKey[]`, line `77` complains that `heldKeys` could still contain `ModifierKey[]`... which, should not be try based on the `hasKeyGroups` ternary.

Not sure how to solve this...

Reply:
You need to do a custom [typeguard](https://basarat.gitbooks.io/typescript/docs/types/typeGuard.html#user-defined-type-guards).

```
   function hasKeyGroups(list: ModifierKey[]): list is WhatevertheNestedTypeIs {
      return heldKeys.every(key => Array.isArray(key))
   }
```